tests/MongoDB.Driver.Tests/IAggregateFluentExtensionsTests.cs

```diff
@@ -174,6 +174,17 @@ public void Group_should_generate_the_correct_document_using_expressions()
             AssertLast(subject, expectedGroup);
         }
 
+        [Fact]
+        public void Group_with_document_in_key_should_generate_the_correct_document_using_expressions()
+        {
+            var subject = CreateSubject()
+                .Group(x => x, g => new { Name = g.Select(x => x + " " + x.LastName).First() });
```

Comment:
I don't understand this part

    g.Select(x => x + " " + x.LastName)

`x` is a `Person` and I don't understand how you can use the `+` operator on `Person` and `string`.

Reply:
I still haven't found any documentation on why this compiles but I have managed to figure out that it is roughly equivalent to:

    g.Select(x => x?.ToString() + " " + x.LastName)

But... `x.ToString()` is meaningless server-side so there is nothing that it could be translated into.

Reply:
This appears to be handled by the compiler. In other words, there is no `String` operator that looks like:

    public static string operator+(object x, string y)

The compiler itself decides to call `x?.ToString()`

Reply:
It occurs to me that this odd expression might just be a typo. Did you actually mean to test this:

    .Group(x => x, g => new { Name = g.Select(x => x.FirstName + " " + x.LastName).First() });

i.e. `x.FirstName` instead of just `x`